### ISSUE TITLE
Rumbling trigger motors doesnt work because WGI correlation fails

### ISSUE THREAD

**Issue by selimrecep** [libsdl-org/SDL#5103]
Hi,
I was checking out the library a bit, and found out there was also a "SDL_GameControllerRumbleTriggers" function, but looks like it only works through WGI API, not XInput, so I tried but looks like it doesnt work because for some reason WGI correlation with the controller can't be made. Is this something specific to model of controllers? Or it might be a bug too? Or I did something wrong, I'd like to provide more information but I am a bit lost within the WGI related functions. And I also tried pressing any key, it doesnt correlate, even XInput correlation is successful.

All information I could gather is:

- During the WGI list update, `__FIVectorView_1_Windows__CGaming__CInput__CGamepad_get_Size(gamepad, &num_gamepads)` returns `num_gamepads` as `0`. Which causes to not even try a correlation. 
- I also realised the previous line I was talking about (where num_gamepads is acquired from WGI API), the line is being called twice, so when I set a breakpoint at the line and breaks at there, and I keep spamming buttons and moving some joysticks on controller, but at the same time continuing the code process, I can see at the second time the line runs `num_gamepad` becomes `1`. But still correlation can't be made due to some condition failure that I couldn't find out yet.
- I also found out at the time of writing sometimes (maybe rarely) `num_gamepads` is `1` without pressing anybutton at the time of start. But still no correlation

My controller is "Xbox Wireless Controller(9thGen, series controller).

**Comment by selimrecep** [comment]
Also for some reason, at `RAWINPUT_GuessWindowsGamingInputSlot` never `match_count == 1 && state->any_data` is `1`, like it is either match_count or state->any_data is `1`, I am still trying to find the root of the issue

**Comment by selimrecep** [comment]
I finally found the issue. And I think this will be really helpful for others as well. Even I dont know in deep, `Windows.Gaming.Input` API definitely doesnt work really well when I dont create a window. Normally, I would just do `SDL_Init(SDL_INIT_VIDEO | SDL_INIT_GAMECONTROLLER);`. But this isn't enough, WGI doesn't correlate at the start of app even though XInput does (when pressing any button). As the link I posted above, the main reason it doesnt work is window is needed to be in focus which isn't something represented in a console application...

I simply put 
```
	window = SDL_CreateWindow(
		"An SDL2 window",                  // window title
		SDL_WINDOWPOS_UNDEFINED,           // initial x position
		SDL_WINDOWPOS_UNDEFINED,           // initial y position
		640,                               // width, in pixels
		480,                               // height, in pixels
		SDL_WINDOW_OPENGL                  // flags - see below
	);
```
and everything working as intended! I don't know if there is a way to use features that use WGI without creating window, but this kinda gave me headache for 2 days, I hope this will be useful to others
https://stackoverflow.com/a/57794492/13943297
